OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -820,12 +820,10 @@ public void ResolveOrder(Actor self, Order order)
 			}
 			else if (order.OrderString == "ReturnToBase" && rearmableInfo != null && rearmableInfo.RearmActors.Any())
 			{
-				UnReserve();
```

Comment:
I think it would be safer to keep this unconditional UnReserve, in case a queued RTB is cancelled before running.

Reply:
Removing this line entirely seems to have been an oversight, but shouldn't it be conditional on not being queued? That is how it works for the other orders as well. I have added this in the new commit.

Reply:
I'm not convinced here - IMO it doesn't make sense to hold the reservation and block other actors from using the airfield until the current activities are completed.  Do we know why the other orders do it like this?

Reply:
The queued unreserve was introduced in #8886.  There is no explicit mention of the reservation, so it may not have been intentional.

Reply:
Hmm, it may be because of the `TakeOff` order in `UnReserve`? Because it only checks the height at the moment the order is given, not when the queued order is executed this may lead to odd behaviour.

Reply:
I see. That's a rather ugly hack.

Lets take this as it is then, and look into improving the `UnReserve` behaviour in a future PR.